### ISSUE TITLE
I believe there is a systemic issue with variable sanitation 

### ISSUE THREAD

**Issue by ns-rlewkowicz** [kubevirt/kubevirt#10420]
The first instance is here:
https://github.com/kubevirt/kubevirt/issues/10401

The TLDR is that the code creates the dom with basic identifiers along side the user defined name. If you use the same name (you have no way of knowing what that name is) things break silently. 

I've run into just another silent issue of naming a filesystem `foo-config`. Suddenly I have no network interface and the 50-cloudinit file is not created. Naming it `foo-cm` vs `foo-config` and it all works again. 

I have to guess it's a similar issue. Something on the dom uses `-config` and there's a text manipulation that is not tight enough. 

The first issue above is verifiable and reproducible, I'll try to find the time to target this similarly. I don't have a lot of bandwidth right now.

**Comment by EdDev** [comment]
This seems to be a meta-issue.
It will preferable to handle one issue at a time, opening one per discovered problem, if it exist.

Consider closing this one.

**Comment by xpivarc** [comment]
Hi @ns-rlewkowicz 
could you share a VMI yaml?

**Comment by EdDev** [comment]
> Hi @ns-rlewkowicz
> could you share a VMI yaml?

This should help: https://github.com/kubevirt/kubevirt/issues/10401
